docs/design/docker-for-binaries/readme.md

```diff
@@ -0,0 +1,114 @@
+# Docker for Binaries
+
+# User Story
+
+As an Ed-Fi system administrator, I want to run the Ed-Fi ODS Admin Api in a container.
+
+# Acceptance Criteria
+
+- Built using the same NuGet packages that the PowerShell installer would use.
+- Auto-published with the latest version available.
+- Can use a pre-built Admin database image containing Admin API database schema.
+- Can use an existing Admin database container, and deploy the database schema into it.
+- Can use an existing Admin database instance that is not in a container, and deploy the database schema into it.
+
+# Implementation notes
+
+Today there are three binary-first Dockerfiles in Admin API: Docker/DB-Admin/pgsql/Dockerfile, mssql.Dockerfile, and pgsql.Dockerfile.
+
+The latter two contain the Admin API application runtime, with configuration differences for the two different database providers.
+
+Based on the notes above, this is the support matrix to test:
+
+
+
+| Admin API Host | Database Engine | Database Host | Admin DB Status | Dockerfiles
+| -----------    | -----------     |-----------    |-----------      |-----------                        |
+| Docker         | PostgreSQL      | Docker        |New              |pgsql/Dockerfile, pgsql.Dockerfile |
+| Docker         | PostgreSQL      | External Host |New              |pgsql.Dockerfile                   |
+| Docker         | PostgreSQL      | Docker        |Already exists   |pgsql.Dockerfile                   |
+| Docker         | PostgreSQL      | External Host |Already exists   |pgsql.Dockerfile                   |
+| Docker         | MSSQL           | External Host |New              |mssql.Dockerfile                   |
+| Docker         | MSSQL           | External Host |Already exists   |mssql.Dockerfile                   |
+
+We do not have rows for MSSQL in Docker because, while it can be done, the Alliance does not provide those images due to licensing restrictions.
+
+The dockerfiles need to accept an argument with the version number, by default will be "latest" in order to get latest version available.
```

Comment:
@lchavarriaga I don't think "latest" will work for NuGet packages, unless you write a script to query the NuGet feed for the available versions, sort, and find latest. In fact, I'm pretty sure that I have written such a script. I will try to find that and link to it in another comment. That script would then need to be run from inside the container, or it could be run outside the container before executing `docker build`.

Reply:
You can copy `Get-NugetPackage` and `Invoke-SemanticSort` from https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/main/eng/bulkLoad/modules/Package-Management.psm1